awx/main/analytics/subsystem_metrics.py

```diff
@@ -70,10 +80,9 @@ def decode_value(self, value):
             return 0
 
     def store_value(self, conn):
-        # do not set value if it has not changed since last time this was called
-        if self.current_value is not None:
+        if self.metric_has_changed:
             conn.hset(root_key, self.field, self.current_value)
-            self.current_value = None
+            self.metric_has_changed = False
 
 
 class SetFloatM(SetIntM):
```

Comment:
Could you add a doc string to the SetFloatM and SetIntM classes to help in the future distinguish between when one would be used vs the FloatM and IntM classes? The names don't communicate enough context. From what I gather w/ @rebeccahhh's PR the SetFloatM and SetIntM are necessary to use if you plan to completely wipeout a metric each time you do push it to redis. The IntM and FloatM `.set` methods we behaving differently when we were experimenting but I never got whole gist of whats going on (they were incrementing periodically in a weird way)

Reply:
my plan is to write up docs/subsystem_metrics.md to explain how the subsystem metrics works